pilot/cmd/istioctl/inject.go

```diff
@@ -18,32 +18,64 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"io/ioutil"
 	"os"
+
 	// TODO(nmittler): Remove this
+	"github.com/ghodss/yaml"
 	_ "github.com/golang/glog"
 	"github.com/spf13/cobra"
-	"k8s.io/api/core/v1"
 
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/pilot/cmd"
+	"istio.io/istio/pilot/model"
 	"istio.io/istio/pilot/platform/kube"
 	"istio.io/istio/pilot/platform/kube/inject"
 	"istio.io/istio/pkg/log"
 	"istio.io/istio/pkg/version"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const configMapKey = "mesh"
+
+func getMeshConfigFromConfigMap(kubeconfig string) (*meshconfig.MeshConfig, error) {
+	_, client, err := kube.CreateInterface(kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+	config, err := client.CoreV1().ConfigMaps(istioNamespace).Get(meshConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not find %q ConfigMap in %q namespace", meshConfigMapName, istioNamespace)
+	}
+	// values in the data are strings, while proto might use a
+	// different data type.  therefore, we have to get a value by a
+	// key
+	yaml, exists := config.Data[configMapKey]
+	if !exists {
+		return nil, fmt.Errorf("missing configuration map key %q", configMapKey)
+	}
+	return model.ApplyMeshConfigDefaults(yaml)
+}
+
 var (
-	hub               string
-	tag               string
-	sidecarProxyUID   int64
-	verbosity         int
-	versionStr        string // override build version
-	enableCoreDump    bool
-	meshConfigMapName string
-	imagePullPolicy   string
-	includeIPRanges   string
-	debugMode         bool
+	hub             string
+	tag             string
+	sidecarProxyUID uint32
```

Comment:
what was the issue with 64 bits?

Reply:
@cmluciano noted that UID wouldn't be negative. I changed to unsigned and also 32bit, but I'll change back to 64-bit since it doesn't really hurt.